src/checkout/components/CheckoutProvider.tsx

```diff
@@ -241,10 +262,13 @@ export const CheckoutProvider: FunctionComponent<PropsWithChildren<
 CheckoutProvider.propTypes = {
   stripe: PropTypes.any,
   options: PropTypes.shape({
-    fetchClientSecret: PropTypes.func.isRequired,
-    elementsOptions: PropTypes.object as any,
+    clientSecret: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.instanceOf(Promise),
+    ]).isRequired,
+    elementsOptions: PropTypes.object,
   }).isRequired,
-};
+} as PropTypes.ValidationMap<CheckoutProviderProps>;
```

Comment:
I have to `as` here because propTypes doesn't support `PropTypes.instanceOf(Promise<string>)`